### PR TITLE
Feature/add resource bundle

### PR DESCRIPTION
**Pull request by TroubleMakerBen**
Creates MobileCenterUpdatesResources.bundle that contains localizations. It contains 1 string for testing for now, new strings can be added easily. The method to read localized strings is part of the update module for now but we can move this to Core at some point.
The most important thing to note is that the bundle needs to be copied and added to the project together with `MobileCenterUpdates.framework` by the user for localization to work if the dev does manual integration.
For cocoapods, I have added the necessary subspec but commented it out for later testing.

### CODE REVIEW

File: MobileCenter.podspec [hunk 1]
```diff
@@ -47,5 +47,15 @@ Pod::Spec.new do |s|
       ss.vendored_frameworks = "MobileCenter-SDK-iOS/MobileCenterCrashes.framework"
   end
 
+#  s.subspec 'MobileCenterUpdates' do |ss|
```
Comment (jaelim-ms) [inline MobileCenter.podspec]: It is commented out. Why?

File: MobileCenter.podspec [hunk 2]
```diff
@@ -47,5 +47,15 @@ Pod::Spec.new do |s|
...
+#   ss.frameworks = 'CoreTelephony', 'Foundation'
+#   ss.dependency 'MobileCenter/MobileCenter'
+#  ss.resource_bundle = { 'MobileCenterUpdatesResources' => ['MobileCenter-SDK-iOS/MobileCenterUpdatesResources.bundle/*.lproj'] }
+# end
+
```
Comment (jaelim-ms) [inline MobileCenter.podspec]: And unnecessary lines.

File: MobileCenterUpdates/MobileCenterUpdates/MSUpdateUtil.h [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#import <Foundation/Foundation.h>
```
Comment (jaelim-ms) [inline MobileCenterUpdates/MobileCenterUpdates/MSUpdateUtil.h]: File name. *MSUpdate**s**Util*?

File: MobileCenterUpdates/MobileCenterUpdates/MSUpdates.m [hunk 4]
```diff
@@ -335,8 +340,8 @@ - (void)handleUpdate:(MSReleaseDetails *)details {
                MS_DEVICE.systemVersion);
     return;
   }
-
```
Comment (jaelim-ms) [inline MobileCenterUpdates/MobileCenterUpdates/MSUpdates.m]: Revert this change and two lines below.

File: MobileCenterUpdates/MobileCenterUpdates/MSUpdates.m [hunk 5]
```diff
@@ -439,4 +444,8 @@ - (void)openUrl:(NSURL *)url {
   }
 }
 
+- (MSInitializationPriority)initializationPriority {
```
Comment (jaelim-ms) [inline MobileCenterUpdates/MobileCenterUpdates/MSUpdates.m]: It has been moved to *MSServiceAbstract*. Remove IT.

File: MobileCenterUpdates/MobileCenterUpdatesTests/MSUpdateUtilTests.m [hunk 6]
```diff
@@ -0,0 +1,39 @@
+//
```
Comment (jaelim-ms) [inline MobileCenterUpdates/MobileCenterUpdatesTests/MSUpdateUtilTests.m]: We don't need this block. 😛

Comment (jaelim-ms) [inline MobileCenterUpdates/MobileCenterUpdatesTests/MSUpdateUtilTests.m]: If you don't have any implementation of this class, let's remove it. It can come later with actual test.

File: Puppet/Puppet.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -25,6 +25,7 @@
 		B21B26C71DDE490400FF0378 /* MSCrashesDetailViewController.m in Sources */ = {isa = PBXBuildFile; fileRef = B21B26C61DDE490400FF0378 /* MSCrashesDetailViewController.m */; };
 		B27397A41DE3D27700AC7761 /* CrashLibIOS.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = B27397A31DE3D27700AC7761 /* CrashLibIOS.framework */; };
 		B27397A51DE3D27700AC7761 /* CrashLibIOS.framework in Embed Frameworks */ = {isa = PBXBuildFile; fileRef = B27397A31DE3D27700AC7761 /* CrashLibIOS.framework */; settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
+		B296D4211E5F5DEF0089F013 /* MobileCenterUpdatesResources.bundle in Resources */ = {isa = PBXBuildFile; fileRef = B2C070CB1E5F59A90076D6A9 /* MobileCenterUpdatesResources.bundle */; };
```
Comment (jaelim-ms) [inline Puppet/Puppet.xcodeproj/project.pbxproj]: I hope this is only for Puppet what uses not frameworks but libraries .

Comment (TroubleMakerBen) [inline Puppet/Puppet.xcodeproj/project.pbxproj]: I'm not getting what you are saying. The bundle needs to be added independently from the `.frameworks`.

File: MobileCenter.podspec (hunk 1 above)
Comment (TroubleMakerBen) [inline MobileCenter.podspec]: We need to create an internal release for the podspec and doing it all locally is simply a task of it's own and is out of scope of this task imho. I'm adding a task for this for next week.

File: MobileCenterUpdates/MobileCenterUpdates/MSUpdates.m (hunk 4 above)
Comment (TroubleMakerBen) [inline MobileCenterUpdates/MobileCenterUpdates/MSUpdates.m]: Not sure what needs to be changed here

File: MobileCenterUpdates/MobileCenterUpdates/MSUpdateUtil.h [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+/*
+ * Return the application's main bundle.
```
Comment (clpolet) [inline MobileCenterUpdates/MobileCenterUpdates/MSUpdateUtil.h]: I think it returns the updates' module's bundle not the app bundle If I understand correctly.

File: Puppet/Puppet.xcodeproj/project.pbxproj (hunk 7 above)
Comment (jaelim-ms) [inline Puppet/Puppet.xcodeproj/project.pbxproj]: Does that mean customers should add this in their applications?

File: MobileCenterUpdates/MobileCenterUpdates/MSUpdates.m (hunk 4 above)
Comment (jaelim-ms) [inline MobileCenterUpdates/MobileCenterUpdates/MSUpdates.m]: I mean you are removing an empty line above the comment and adding it below the comment.

File: Puppet/Puppet.xcodeproj/project.pbxproj (hunk 7 above)
Comment (TroubleMakerBen) [inline Puppet/Puppet.xcodeproj/project.pbxproj]: They need to, yes. Same with HockeyApp.

File: MobileCenterUpdates/MobileCenterUpdates/MSUpdates.m (hunk 4 above)
Comment (TroubleMakerBen) [inline MobileCenterUpdates/MobileCenterUpdates/MSUpdates.m]: The code looks fine in the file.

Comment (jaelim-ms) [inline MobileCenterUpdates/MobileCenterUpdates/MSUpdates.m]: I trust you. 😄

Comment (jaelim-ms) [inline MobileCenterUpdates/MobileCenterUpdates/MSUpdates.m]: Now it shows properly..
